core/src/main/java/com/valtech/aem/saas/core/fulltextsearch/SearchResultsImpl.java

```diff
@@ -0,0 +1,170 @@
+  public static final int DEFAULT_START_PAGE = 0;
+  public static final int DEFAULT_RESULTS_PER_PAGE = 10;
+  public static final String SEARCH_TERM = "q";
+  public static final String QUERY_PARAM_ROWS = "rows";
+  public static final String QUERY_PARAM_LANGUAGE = "language";
+  public static final String DEFAULT_LANGUAGE = "en";
+  public static final String I18N_KEY_LOAD_MORE_BUTTON_LABEL = "com.valtech.aem.saas.core.search.loadmore.button.label";
+
+  @Self
+  private SlingHttpServletRequest request;
+
+  @OSGiService
+  private FulltextSearchService fulltextSearchService;
+
+  @OSGiService
+  private FulltextSearchConfigurationService fulltextSearchConfigurationService;
+
+  @OSGiService
+  private I18nProvider i18nProvider;
+
+  @ScriptVariable
+  private Page currentPage;
+
+  @JsonInclude(Include.NON_EMPTY)
+  @Getter
+  private String term;
+
+  @JsonInclude(Include.NON_EMPTY)
+  @Getter
+  private int startPage;
+
+  @JsonInclude(Include.NON_EMPTY)
+  @Getter
+  private int resultsPerPage;
+
+  @JsonInclude(Include.NON_NULL)
+  @Getter
+  private List<Result> results;
+
+  private int configuredResultsPerPage;
+
+  private I18n i18n;
+
+  @PostConstruct
+  private void init() {
+    i18n = i18nProvider.getI18n(request);
+    configuredResultsPerPage = getConfiguredResultsPerPage();
+    RequestParameters requestParameters = new RequestParameters(request);
+    requestParameters.getParameter(SEARCH_TERM).ifPresent(t -> {
+      term = t;
+      startPage = requestParameters.getParameter(QUERY_PARAM_START)
+          .map(s -> new StringToInteger(s).asInt())
+          .map(OptionalInt::getAsInt)
+          .orElse(DEFAULT_START_PAGE);
+      resultsPerPage = resolveResultsPerPage();
+      SearchConfiguration searchConfiguration = request.getResource().adaptTo(ConfigurationBuilder.class)
+          .as(SearchConfiguration.class);
+      FulltextSearchGetRequestPayload fulltextSearchGetRequestPayload =
+          DefaultFulltextSearchRequestPayload.builder(new DefaultTermQuery(term),
+                  new DefaultLanguageQuery(getLanguage()))
+              .optionalQuery(
+                  new PaginationQuery(startPage, resultsPerPage, fulltextSearchConfigurationService.getRowsMaxLimit()))
+              .optionalQuery(new HighlightingTagQuery(Highlighting.HIGHLIGHTING_TAG_NAME))
+              .optionalQuery(FiltersQuery.builder()
+                  .filterEntries(Arrays.stream(searchConfiguration.searchFilters()).collect(Collectors.toMap(
+                      SearchFilterConfiguration::name, SearchFilterConfiguration::value))).build())
+              .build();
+      results = fulltextSearchService.getResults(searchConfiguration.index(), fulltextSearchGetRequestPayload)
+          .map(FulltextSearchResults::getResults).orElse(
+              Collections.emptyList());
+    });
+  }
+
+  private int resolveResultsPerPage() {
+    return new RequestParameters(request).getParameter(QUERY_PARAM_ROWS)
```

Comment:
This is only for demo purpose for the **load more** feature. Since the page is completely reloaded with a **load more** action (it was decided that ajax calls are not in the scope of the BE ticket). This snippet increases the **rows** query parameter by the configured value for results per page.

Reply:
In such cases additional comments/todos should be added for temporary code with links to the ticket that should fix that